### PR TITLE
Dropdown: Custom render options for multiselect - Bug fix #3571 

### LINKED ISSUE #3571: Dropdown in MultiSelect mode `onRenderOption` doesnt work.
The onRenderOption doesnt work in drop down multiselect mode.

### PR DESCRIPTION
**Pull request by rohitjindal18**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #3571 
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Dropdown used Checkbox component in case of multiselect mode, and  CommandButton component in case of non multiselect mode.

onRenderOption was not getting used in Dropdown multiselect Mode, onRenderOption was being passed as Checkbox children, but Checkbox was not making use of its children, instead it accepts a onRenderLabel prop to render custom JSX.
Passed onRenderOption in onRenderLabel props

#### Focus areas to test

Pass multiSelect as true and onRenderOption to render custom JXS for Dropdown options.

### CODE REVIEW

**Comment by rohitjindal18** [comment]
@jakob101 Can you please review the changes

**Comment by jakob101** [comment]
Yeah, give me a sec ;)

________________________________

**Comment by rohitjindal18** [comment]
@jakob101 Are any changes required in the code?

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -514,6 +514,7 @@ export class Dropdown extends BaseComponent<IDropdownInternalProps, IDropdownSta
               onMouseMove: this._onItemMouseMove.bind(this, item)
             } }
             label={ item.text }
+            onRenderLabel={ this._renderCustomLabel.bind(this, item) }
```
Comment (erichdev) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: Use `@autobind`. Also, `_onRenderLabel` might be a better name to match existing naming conventions in this file

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -843,5 +843,8 @@ export class Dropdown extends BaseComponent<IDropdownInternalProps, IDropdownSta
       });
     }
   }
-
-}
+  private _renderCustomLabel(item: IDropdownOption) {
```
Comment (erichdev) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: Minor -- I don't think Fabric has any style guidelines on this, but I'd move this method up with all the other render functions, around line 545

**Comment by manishgarg1** [comment]
Hi @rohitjindal18, this has been open for a while. The comments are simple to fix. Do you still want to merge this?

**Comment by rohitjindal18** [comment]
@manishgarg1 Have fixed the review comments, one  screener check is not passing, saw the dashboard, there is visual change in multi select example for dropdown, but doesn't look like an issue.

**Comment by dzearing** [comment]
We need a change file here.

You can run `npm run change` from the root to generate one.

**Comment by rohitjindal18** [comment]
Added change file @dzearing @manishgarg1

**Pull request merged**